File: design-pattern-go/src/patterns/behavioral/mediator/ControlTower.go
package mediator

type ControlTower struct {
}

func NewControlTower() *ControlTower {
	return &ControlTower{}
}

func (ct *ControlTower) Notify(_ Airship, event string) string {
	switch event {
	case "beforeLand":
		return "preparing route to land"
	case "afterLand":
		return "preparing for passengers unboarding"
	case "afterLiftOff":
		return "cleaning up route to other airship"
	default:
		return "preparing route to lift off"
	}
}

File: design-pattern-jvm/app/src/main/kotlin/design/pattern/kotlin/patterns/behavioral/mediator/ControlTower.kt
package design.pattern.kotlin.patterns.behavioral.mediator

open class ControlTower : Mediator {
    override fun notify(airship: Airship, event: String): String {
        return when (event) {
            "beforeLand" -> "preparing to land"
            "afterLand" -> "preparing for passengers boarding"
            "afterLiftOff" -> "cleaning up route to other airship"
            else -> "preparing route to lift off"
        }
    }
}